cmd/rook/ceph/osd.go

```diff
@@ -192,6 +198,10 @@ func prepareOSD(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	if err := readCephSecret(path.Join(mon.CephSecretMountPath, mon.CephSecretFilename)); err != nil {
+		rook.TerminateFatal(err)
+	}
+
```

Comment:
Do we really need these? If the deployment defines an env var, then that version will reference the env var, and if it defines the file, then it will use the file, right? Is there some other behavior I've forgotten where Rook needs to read that info?

Reply:
For the mgr side car and the osd prepare job, their pod specs are controlled by the operator of the same version, so we know that those will be using the secret file. But the osd purge job could have a spec with the env vars, or it could have the secret mounted, depending on the version. So this helper will use the fallback to the env var, really just to cover that osd purge job case.